src/treemodel/selection.js

```diff
@@ -83,11 +86,26 @@ export default class Selection {
 	}
 
 	/**
+	 * Flag indicating whether the selection has any range in it.
+	 *
+	 * @readonly
+	 * @type {Boolean}
+	 */
+	get hasAnyRange() {
```

Comment:
I think that `.hasAnyRange` is more precise. If I've seen `isEmpty` I'd think to myself whether this mean "has no ranges" or "has no nodes / collapsed". I've changed it to `hasAnyRanges` (added "s").